src/library/scala/jdk/javaapi/FutureConverters.scala

```diff
@@ -72,7 +72,12 @@ object FutureConverters {
       case f: Future[T @unchecked] => f
       case _ =>
         val p = new P[T](cs)
-        cs whenComplete p
+        val future = cs.toCompletableFuture
```

Comment:
> A CompletionStage implementation that does not choose to interoperate with others may throw UnsupportedOperationException

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletionStage.html#toCompletableFuture--

Should that case be considered?

Reply:
done, thanks